### DISCUSSION TITLE
Disabled dom diffing?

### DISCUSSION THREAD

**Discussion by mityukov** [livewire/livewire#2487] [3. Feature ideas]
Hi, is there an option to disable DOM-diffing completely for selected component/element?

What I mean: there is a rather complex component where I'm stumbling on DOM diffing issues all the time (everything is covered with `wire:key`'s, but still…) And the final nail: that component, being rather complex, is not too often redrawn, so, the speed of redrawing is not that important. Stability is, however…

Maybe, a tag like `wire:no-diffing`?

**Comment by nuernbergerA** [comment]
isn't that what `wire:ignore` does?
https://laravel-livewire.com/docs/2.x/alpine-js
search for "Ignoring DOM-changes"

**Reply by mityukov** [reply]
No, I think not. `wire:ignore` tells Livewire to stop tracking changes inside specified element.

What I need — is to tell Livewire to "redraw the whole widget after each Livewire request". Meaning, not even to try looking for divs that may be preserved. Because that economy in DOM manipulations (which not every eye would actually catch or care)  is achieved at the cost of work/time/money, when the developer has to fight with `wire:key` attributes to make sure nothing gets stuck.

Ideally: to have an ability to make it in "full on refresh mode". Then, **IF** it feels sluggish — do the optimization by adding some keys that cache DOM.

**Comment by joshhanley** [comment]
@mityukov have a look at this package by @stancl https://github.com/LeanAdmin/wire-replace

**Reply by mityukov** [reply]
Thank you! If I add this attribute to the, let say, "Whole Table Container" — will it replace the whole HTML with actual data? Or, should I only use this only in specific "cells" where `wire:key` doesn't work?

**Reply by mityukov** [reply]
as usual: Laravel people are the best in naming things!

**Reply by joshhanley** [reply]
No worries.

Haha ah naming things!

Up to you, depends on what you need. You can apply to the whole table or just individual cells. If the table is just data, then maybe apply to the whole table, but if the table contains buttons and things which will be bound to JS, maybe better to just apply to specific cells, saves re-creating and re-binding buttons every time it loads.
